tests/app/test_users.py

```diff
@@ -181,6 +190,20 @@ def test_return_400_for_invalid_user_json(self):
                 'users': {
                     'email_address': 'joeblogs@email.com',
                     'password': '',
+                    'role': 'buyer',
+                    'name': 'joe bloggs'}}),
+            content_type='application/json')
+
+        assert_equal(response.status_code, 400)
+
+    def test_return_400_for_invalid_user_role(self):
```

Comment:
We only validate that 'buyer' is valid and 'invalid' is not valid. We never test that 'supplier' and 'admin' are valid roles.